java/code/src/com/suse/manager/maintenance/MaintenanceManager.java

```diff
@@ -777,90 +640,67 @@ public boolean isActionInMaintenanceWindow(Action action, MaintenanceSchedule sc
-     * @param systemIds the set of {@link Server} IDs
-     * @throws PermissionException if the user does not have access to given servers
-     * @throws IllegalArgumentException if systems have pending maintenance-only actions
-     * @return the number of involved {@link Server}s
+     * @param server the server to check
+     * @return true when the action is inside of a maintenance window, otherwise falsegg
      */
-    public int assignScheduleToSystems(User user, MaintenanceSchedule schedule, Set<Long> systemIds) {
-        ensureOrgAdmin(user);
-        ensureSystemsAccessible(user, systemIds);
-        ensureScheduleAccessible(user, schedule);
-
-        Set<Long> withMaintenanceActions = ServerFactory.filterSystemsWithPendingMaintOnlyActions(systemIds);
-        if (!withMaintenanceActions.isEmpty()) {
-            throw new IllegalArgumentException("Systems have pending maintenance-only actions:" +
-                    withMaintenanceActions);
-        }
-
-        return ServerFactory.setMaintenanceScheduleToSystems(schedule, systemIds);
+    public boolean isSystemInMaintenanceMode(MinionServer server) {
+        return server.getMaintenanceScheduleOpt()
+                .map(schedule -> !getCalendarForNow(schedule).isEmpty())
+                .orElse(true);
     }
 
     /**
-     * Retract {@link MaintenanceSchedule} from given set of {@link Server}s.
+     * Given a list of minions, sorts by maintenance mode status, logs skipped minions
      *
-     * @param user the user
-     * @param systemIds the set of {@link Server} IDs
-     * @throws PermissionException if the user does not have access to given servers
-     * @return the number of involved {@link Server}s
-     */
-    public int retractScheduleFromSystems(User user, Set<Long> systemIds) {
-        ensureOrgAdmin(user);
-        ensureSystemsAccessible(user, systemIds);
+     * @param minions servers to check
+     * @return List of minions in maintenance mode
+     */
+    public List<Long> systemIdsMaintenanceMode(List<MinionServer> minions) {
+        Set<MaintenanceSchedule> schedulesInMaintMode = minions.stream()
+                .flatMap(minion -> minion.getMaintenanceScheduleOpt().stream())
+                .distinct()
+                .filter(sched -> !getCalendarForNow(sched).isEmpty())
+                .collect(Collectors.toSet());
+
+        List<Long> minionsInMaintMode = minions.stream()
+                .filter(minion -> minion.getMaintenanceScheduleOpt()
+                .map(sched -> schedulesInMaintMode.contains(sched)) // keep minions that have maintenance mode
+                .orElse(true)) // or that have no maintenance schedule whatsoever
+                .map(minion -> minion.getId())
+                .collect(toList());
 
-        return ServerFactory.setMaintenanceScheduleToSystems(null, systemIds);
+         List<MinionServer> logList = minions.stream()
+                 .filter(m -> !minionsInMaintMode.contains(m))
```

Comment:
This give a warning in eclipse as "m" is a full minion object and we check a `List<Long>.contains(Minion)`

Reply:
This is an inherited code, but I'll fix that anyway. Good catch!